diff --git a/out/node-file-trace.d.ts b/out/node-file-trace.d.ts
index a922f41adbaa4d4c2829cdda206414eeb34b345f..7882694c147b8fb9bea4571e6d02a71d3d789f36 100644
--- a/out/node-file-trace.d.ts
+++ b/out/node-file-trace.d.ts
@@ -24,7 +24,7 @@ export declare class Job {
     reasons: NodeFileTraceReasons;
     private cachedFileSystem;
     private remappings;
-    constructor({ base, processCwd, exports, conditions, exportsOnly, paths, ignore, log, mixedModules, ts, analysis, cache, fileIOConcurrency, depth, }: NodeFileTraceOptions);
+    constructor({ base, processCwd, exports, conditions, exportsOnly, paths, ignore, log, mixedModules, ts, analysis, cache, fileIOConcurrency, depth, web, }: NodeFileTraceOptions);
     addRemapping(path: string, dep: string): void;
     readlink(path: string): Promise<string | null>;
     isFile(path: string): Promise<boolean>;
diff --git a/out/node-file-trace.js b/out/node-file-trace.js
index ab58862c98f3f024d74668495b00a98a95f76f03..0cf470015d63ede233cd408586830eaff1675718 100644
--- a/out/node-file-trace.js
+++ b/out/node-file-trace.js
@@ -92,7 +92,7 @@ class Job {
     reasons = new Map();
     cachedFileSystem;
     remappings = new Map();
-    constructor({ base = process.cwd(), processCwd, exports, conditions = exports || ['node'], exportsOnly = false, paths = {}, ignore, log = false, mixedModules = false, ts = true, analysis = {}, cache, 
+    constructor({ base = process.cwd(), processCwd, exports, conditions = exports || ['node'], exportsOnly = false, paths = {}, ignore, log = false, mixedModules = false, ts = true, analysis = {}, cache, web = false, 
     // we use a default of 1024 concurrency to balance
     // performance and memory usage for fs operations
     fileIOConcurrency = 1024, depth = Infinity, }) {
@@ -139,6 +139,7 @@ class Job {
         this.log = log;
         this.depth = depth;
         this.mixedModules = mixedModules;
+        this.web = web;
         this.cachedFileSystem = new fs_1.CachedFileSystem({ cache, fileIOConcurrency });
         this.analysis = {};
         if (analysis !== false) {
diff --git a/out/resolve-dependency.js b/out/resolve-dependency.js
index fc0b8a296c1428653326f9f7f74f4b4f3346a5b5..6dd28e3b3f4913f7a619bfaf863baea5456cd134 100644
--- a/out/resolve-dependency.js
+++ b/out/resolve-dependency.js
@@ -72,9 +72,10 @@ async function resolveDir(path, parent, job) {
     if (!(await job.isDir(path)))
         return;
     const pkgCfg = await getPkgCfg(path, job);
-    if (pkgCfg && typeof pkgCfg.main === 'string') {
-        const resolved = (await resolveFile((0, path_1.resolve)(path, pkgCfg.main), parent, job)) ||
-            (await resolveFile((0, path_1.resolve)(path, pkgCfg.main, 'index'), parent, job));
+    if (pkgCfg && ((job.web && (typeof pkgCfg.browser === 'string' || typeof pkgCfg.module === 'string')) || typeof pkgCfg.main === 'string')) {
+        const fieldValue = job.web ? (typeof pkgCfg.browser === 'string' ? pkgCfg.browser : pkgCfg.web): pkgCfg.main;
+        const resolved = (await resolveFile((0, path_1.resolve)(path, fieldValue), parent, job)) ||
+            (await resolveFile((0, path_1.resolve)(path, fieldValue, 'index'), parent, job));
         if (resolved) {
             await job.emitFile(path + path_1.sep + 'package.json', 'resolve', parent);
             return resolved;
diff --git a/out/types.d.ts b/out/types.d.ts
index 819c25a6bb5475162fc0069eb35d62fb0519f79b..4e37e4912d4c0ab80262a8e9e3edcefd5093e9fa 100644
--- a/out/types.d.ts
+++ b/out/types.d.ts
@@ -49,6 +49,7 @@ export interface NodeFileTraceOptions {
     resolve?: (id: string, parent: string, job: Job, cjsResolve: boolean) => Promise<string | string[]>;
     fileIOConcurrency?: number;
     depth?: number;
+    web?: boolean;
 }
 export type NodeFileTraceReasonType = 'initial' | 'resolve' | 'dependency' | 'asset' | 'sharedlib';
 export interface NodeFileTraceReasons extends Map<string, {
